content/en/blog/2025/ottl-contexts-just-got-easier.md

```diff
@@ -0,0 +1,131 @@
+structures. This style support statements from multiple contexts and does not
+require grouping them separately.
+
+To illustrate this, compare the following configuration:
+
+```yaml
+metric_statements:
+  - context: resource
+    statements:
+      - keep_keys(attributes, ["host.name"])
+  - context: metric
+    statements:
+      - set(description, "Sum") where type == "Sum"
+      - convert_sum_to_gauge() where name == "system.processes.count"
+  - context: datapoint
+    statements:
+      - limit(attributes, 100, ["host.name"])
+```
+
+With the new flat configuration style, the same logic is expressed more
+concisely as:
+
+```yaml
+metric_statements:
+  - keep_keys(resource.attributes, ["host.name"])
+  - set(metric.description, "Sum") where metric.type == "Sum"
+  - convert_sum_to_gauge() where metric.name == "system.processes.count"
+  - limit(datapoint.attributes, 100, ["host.name"])
+```
+
+This streamlined approach enhances readability and makes configuration more
+intuitive. Please note that all paths in the statements must be prefixed with
+their respective contexts. These prefixes are required for all context-inferred
+configurations and serve as hints for selecting the best match. It also makes
+statements unambiguous and portable between components.
+
+### Structured configuration
+
+The context-inferred structured configuration style closely resembles the
+existing format and allows users to leverage the benefits of context inference
+while providing granular control over statement configurations, such as
+`error_mode` and `conditions`. For example, consider the following
+configuration:
+
+<!-- prettier-ignore-start -->
+```yaml
+metric_statements:
+  - context: datapoint
+    conditions:
+      - resource.attributes["service.name"] == "my.service"
+    statements:
+      - set(metric.description, "counter") where attributes["my.attr"] == "some"
+```
+<!-- prettier-ignore-end -->
+
+It can now be written as:
+
+<!-- prettier-ignore-start -->
+```yaml
+metric_statements:
+  - conditions:
+      - resource.attributes["service.name"] == "my.service"
+    statements:
+      - set(metric.description, "counter") where datapoint.attributes["my.attr"] == "some"
+```
+<!-- prettier-ignore-end -->
+
+In this example, the `context` value is omitted and is automatically inferred to
+`datapoint`, as it is the only context present in the statements that supports
+parsing both `datapoint` and `metric` data.
+
+If we update the above configuration removing the `datapoint` usage:
+
+<!-- prettier-ignore-start -->
+```yaml
+metric_statements:
+  - conditions:
+      - resource.attributes["service.name"] == "my.service"
+    statements:
+      - set(metric.description, "counter")
+```
+<!-- prettier-ignore-end -->
+
+The context inferrer would select the `metric` context instead, since no data
+points are accessed. Although it would be possible to run the statements using
+the `datapoint` context, `metric` is the most efficient option.
+
```

Comment:
Missing here is probably a subsection stating when you should use which form. Users will generally want to know which they should generally prefer, and when.

Reply:
I think some more opinionation is needed, yes. Each of those preludes describes a positive attribute of each mode, but there's no description of when you should use which one.

Reply:
I've added an extra section [here](https://github.com/open-telemetry/opentelemetry.io/pull/6290/files#diff-34aed2660bd751b07ee7b7031b1171d91ab80dc9463584626ee01d70ee7a12f8R127), could you please take a look? Thanks!

Reply:
Awesome, this looks great!